deploy/contracts.go

```diff
@@ -0,0 +1,507 @@
+type syncNeoFSContractPrm struct {
+	logger *zap.Logger
+
+	blockchain Blockchain
+
+	// based on blockchain
+	monitor *blockchainMonitor
+
+	localAcc *wallet.Account
+
+	// address of the NeoFS NNS contract deployed in the blockchain
+	nnsContract util.Uint160
+	systemEmail string
+
+	committee keys.PublicKeys
+
+	// with localAcc signer only
+	simpleLocalActor *actor.Actor
+	// committee multi-sig signs, localAcc pays
+	committeeLocalActor *notary.Actor
+
+	localNEF      nef.File
+	localManifest manifest.Manifest
+
+	// L2 domain name in domainContractAddresses TLD in the NNS
+	domainName string
+
+	// if set, syncNeoFSContract attempts to deploy the contract when it's
+	// missing on the chain
+	tryDeploy bool
+	// 0: committee witness is not needed
+	// WitnessValidators: committee 2/3n+1 with validatorsDeployAllowedContracts
+	// WitnessValidatorsAndCommittee: WitnessValidators + committee n/2+1 with allowed NNS contract calls
+	deployWitness Witness
+	// contracts that are allowed to be called for the validators-witnessed deployment
+	validatorsDeployAllowedContracts []util.Uint160
+	// additional option for unset tryDeploy to specify deployer of the contract
+	// designated globally. Has no effect if tryDeploy is set.
+	designatedDeployer *util.Uint160
+
+	// optional constructor of extra arguments to be passed into method deploying
+	// the contract. If returns both nil, no data is passed (noExtraDeployArgs can
+	// be used).
+	//
+	// Ignored if tryDeploy is unset.
+	buildExtraDeployArgs func() ([]any, error)
+
+	// address of the Proxy contract deployed in the blockchain. The contract
+	// pays for update transactions.
+	proxyContract util.Uint160
+	// set when syncNeoFSContractPrm relates to Proxy contract. In this case
+	// proxyContract field is unused because address is dynamically resolved within
+	// syncNeoFSContract.
+	isProxy bool
+}
+
+// ContractPrm describes parameters required to configure user's contract deployment routine.
+type ContractPrm struct {
+	// Writes progress into the log.
+	Logger *zap.Logger
+	// Particular Neo blockchain instance to be used as NeoFS Sidechain.
+	Blockchain Blockchain
+	// Local process account used for transaction signing (must be unlocked).
+	LocalAccount *wallet.Account
+	// Validator multi-sig account to spread initial GAS to network
+	// participants (must be unlocked).
+	ValidatorMultiSigAccount *wallet.Account
+	// NNSContractAddress contains address of NNS contract in chain.
+	NNSContractAddress util.Uint160
+	// NNSContractAddress contains address of Proxy contract in chain.
+	ProxyContractAddress util.Uint160
+	// SystemEmail which will be set to NNS record after contract deploy.
+	SystemEmail string
+	// DomainName for the deploying contract.
+	DomainName string
+	// NEFFile for deploying contract.
+	NEFFile nef.File
+	// ManifestFile for deploying contract (json).
+	ManifestFile manifest.Manifest
+	// DesignatedDeployer ???
```

Comment:
???

Reply:
Ow, I didn't find an appropriate comment at the moment, but not sure updated one is the best one